### PR TITLE
Fixed some positioning issues of dropdowns and tooltips

### PR DESCRIPTION
**Pull request by msdm**
- Fixed missRight and missLeft calculation bug
- Fixed positioning problems for RTL

### CODE REVIEW

**Comment by gakimball** [comment]
Could you explain in more detail the changes you made to fix these issues?

File: scss/foundation/components/_tooltips.scss [hunk 1]
```diff
@@ -74,11 +74,11 @@ $tooltip-max-width: 300px !default;
         position: absolute;
         top: -($tooltip-pip-size * 2);
         width: 0;
-        #{$default-float}: $tooltip-pip-size;
```
Comment (msdm) [inline scss/foundation/components/_tooltips.scss]: Why do we set the tooltip position based on $default-float value? the positioning is handled in the js file.

File: js/foundation/foundation.tooltip.js [hunk 2]
```diff
@@ -234,8 +234,6 @@
         }).end();
       };
 
-      objPos(tip, (target.offset().top + target.outerHeight() + 10), 'auto', 'auto', target.offset().left);
```
Comment (msdm) [inline js/foundation/foundation.tooltip.js]: This line was redundant and could generate some problems in the calculation of "left" value

**Comment by msdm** [comment]
Sure,
**Tooltips**
The scss assignment of left/right positioning based on $default-float was wrong because ositioning is handled in js. Also the removed line from the associated js file was redundant and could generate some problems in the calculation of "left" value.

**Dropdowns**
The calculation of missRight and missLeft in _base() (lines 296 to 305 of foundation.dropdowns.js) was wrong. Also they were coupled to each other with no reason. Added if-condition in this function is to correctly calculate initial "left" value for RTL.
The changed codes in top() and bottom() were for preventing out of screen dropdowns when either missRight or missLeft is true. Notice that if both missRight and missLeft is true, the default position is selected based on LTR/RTL.
For left() and right(), when missLeft or missRight is true respectively, we just need to call bottom() to fallback to the default position.

**Comment by Behzad63** [comment]
Committed changes have fixed positioning issues in right to left mode.

**Comment by zurbchris** [comment]
We pulled in a change to speed up the repositioning for dropdowns, if you pull that in and resubmit this with those changes, I would like to take a look at it.

**Comment by zurbchris** [comment]
I pulled this down, and while it passes testing, it does some weird things it shouldn't. I've take two screenshots, one from the current Foundation Docs page, one from my local dev environment with your PR. Notice the pip positioning is wrong in the first image. 
screen shot 2015-09-04 at 9 54 18 am
screen shot 2015-09-04 at 9 54 28 am

**Comment by msdm** [comment]
Strange. I tested it again right now and evertything seems work just fine on my local dev env. The wrongly positioned pip in the first screenshot can only happen for RTL pages on small screens (when this.small() returns true) because the current implementation of the dropdowns does not locate the pip in small screens according to the target element position. So, please let me know if this screenshot belongs ot an LTR or RTL page.

**Comment by zurbchris** [comment]
LTR, medium window size.

**Comment by msdm** [comment]
I could not reproduce the case. I even downloaded the corresponding Foundation doc page and added the modified foundation.dropdown.js as the last script file to it but the pip was correctly located in the left side.
I wonder how the pip is in the right side in your screenshot because its location direction is always set according to webpage direction (left for LTR and right for RTL).

**Comment by kball** [comment]
This PR has gone stale and is no longer relevant to master.  Please resubmit against the V5 branch.

**Pull request closed without merging**